### PR TITLE
Implemented initial version with go-github

### PR DESCRIPTION
**Pull request by tboerger**
I will do an initial test now, but it should already look pretty good.

### CODE REVIEW

**Comment by tboerger** [comment]
Finally I would say it's done! It supports GHE as well as I'm using google/go-github as a base library now. It's written entirly in Golang.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Docker image for the Drone NPM plugin
```
Comment (donny-dont) [inline Dockerfile]: Drone GitHub Release plugin

File: DOCS.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Use the GitHub Release plugin for publishing files and artifacts to GitHub
```
Comment (donny-dont) [inline DOCS.md]: Maybe just plugin for publishing and artifcats to GitHub? releases sounds redundant.

File: DOCS.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+* `base_url` - GitHub base URL; only required for GHE
+* `upload_url` - GitHub upload URL; only required for GHE
+
+Sample configuration:
+
+```yaml
+publish:
+  github_release:
+    api_key: my_github_api_key
+    files:
```
Comment (donny-dont) [inline DOCS.md]: files or file? This an array or an array or string?

**Comment by tboerger** [comment]
Fixed the comments

File: DOCS.md (hunk 3 above)
Comment (tboerger) [inline DOCS.md]: files, fixed the above description.

**Pull request merged**